mirrord-layer/src/lib.rs

```diff
@@ -480,6 +527,20 @@ async fn poll_agent(mut pf: Portforwarder, mut receiver: Receiver<HookMessage>)
     let close_file_handler = Mutex::new(Vec::with_capacity(4));
 
     let mut ping = false;
+
+    if ENABLED_OVERRIDE_ENV_VARS.get().is_some() {
```

Comment:
Isn't it enough that you've received the filter_env_vars argument?

Reply:
Alternatively, since you're accessing the configuration here anyway, why not initialise the whole thing here and forgo the filter_env_vars param entirely?